test/reduce.js

```diff
@@ -404,6 +440,21 @@ function to_comment(value) {
     return ("" + value).replace(/\n/g, "\n// ");
 }
 
+function has_exit(fn) {
+    var found = false;
+    var tw = new U.TreeWalker(function(node) {
+        if (found) return found;
+        if (node instanceof U.AST_Exit) {
+            return found = true;
+        }
+        if (node instanceof U.AST_Lambda) {
+            return true; // don't descend into nested functions
+        }
+    });
+    U.walk_body(fn, tw);
```

Comment:
I don't understand why a simple `fn.walk(tw)` won't suffice here?
AFAICT we only need to change `node instanceof U.AST_Lambda` above to `node instanceof U.AST_Scope && node !== fn`, which is a common pattern.

Reply:
I had what you described but thought it was a waste of CPU. The walk_body does exactly what I want without the need for such a check.

Reply:
btw, there are two calls to `walk_body` in `lib/compress.js`.

Reply:
I'll make the change you suggested, but it's less efficient.

Reply:
I think `walk_body()` is an artifact left over from `uglify-es`, as they have that `AST_Arrow` madness.

Reply:
AST_Arrow could have been implemented better in hindsight, but I think there's a legitimate case for `walk_body`. No matter - I made the changes locally pending testing.